src/libcollections/vec_deque.rs

```diff
@@ -2158,6 +2158,18 @@ impl<A: PartialEq> PartialEq for VecDeque<A> {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<A: Eq> Eq for VecDeque<A> {}
 
+#[stable(feature = "vec-deque-partial-eq-slice", since = "")]
+impl<'a, A: PartialEq> PartialEq<&'a [A]> for VecDeque<A> {
```

Comment:
`v[..]` must be enough. Still, less than ideal, but not surprising, since it's generic and thus `&v` can't coerce to anything anyway, it would need an explicit impl for `&Vec<_>`, would it not, to have `d == &v` compile?